hack/externalnode/install-vm.sh

```diff
@@ -141,17 +153,278 @@ EOF
     systemctl status "$ANTREA_AGENT"
 }
 
+# Generic function to execute command with timeout.
+execute_command() {
+    cmd="$1"
+    timeout="$2"
+    echo "Executing $cmd with timeout $timeout at $(date)"
+    timeout --preserve-status --foreground $timeout $cmd
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        echoerr "$cmd failed with status $ret at $(date), exiting"
+        exit 2
+    fi
+}
+
+# Install docker as container runtime.
+install_container_runtime() {
+    echo "Installing container runtime on $OS_NAME"
```

Comment:
Shall we require docker is pre-installed as a prerequisite?  I don't think docker installation is in the scope of antrea. I think it makes more sence to add a parameter to let the caller deside if or not docker installation is expected. If it is not expected, we should use the existing docker utility on the VM directly, or return error if it is not existing. Only when user explicity expects to install docker, we should do the installation? 

@tnqn @jianjuns What is your thought?